### PR TITLE
Allow chained actions for `[#item_title]`

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
### Summary

Added the option to have chained actions for Post Title field in the Map Content on the tag `[#item_title]`. ( the process is similar to the one from post content)

Changes:
- added tagify with the custom format for Post Title field that allow serialization for chained actions.
- added `[#item_title]` in the list of allowed tags that can use the chained actions.
- updated the data processing for `$post_title` to account for the new format.

If the user had some tags already in the title field, they will appear normal since they are saved like `[#item_title], [#item_source]`.

The workflow:
1. The import UI encodes all the chained actions data inside the Tagify action. (`import-metabox-edit.js`)
2. On the import process, the field data is decoded and extracted from the tagify structure. (`feedzy-rss-feeds-import.php`)
3. The structure is a serialized array with all the actions. The data field will be a mix of tags and normal text (if present).
4. The tags are extracted from the entire field data. Then they are deserialized and processed.
5. The result of the process is replaced in the field data (swaps the result with serialized data in the string).

> [!NOTE]
> I did some code refactoring on the function and variable names so that the above process is reflected in their names.

> [!NOTE]
> I removed the code from [PR](https://github.com/Codeinwp/feedzy-rss-feeds/pull/905) since they conflict. This version is more advance. 

### Will affect visual aspect of the product

NO

### Screenshots 

https://github.com/Codeinwp/feedzy-rss-feeds/assets/17597852/28367d6e-9b96-4e92-8e3e-09400780531f

## Check before Pull Request is ready:

* [x] I have [written a test](CONTRIBUTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](CONTRIBUTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](CONTRIBUTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](CONTRIBUTING.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](CONTRIBUTING.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

Closes https://github.com/Codeinwp/feedzy-rss-feeds-pro/issues/645

### CODE REVIEW

File: includes/views/js/import-metabox-edit.js [hunk 1]
```diff
@@ -550,6 +545,39 @@
...
 			}
 		} );
+		
+		$( '.fz-input-tagify[name="feedzy_meta_data[import_post_title]"]:not(.fz-tagify-image)' ).tagify( {
+			mode: 'mix',
+			editTags: false,
+			templates: {
+				tag: function( tagData ) {
+					try{
+						console.log(tagData);
```
Comment (preda-bogdan) [inline includes/views/js/import-metabox-edit.js]: I think this was left from debugging, we can remove this.

File: includes/views/js/import-metabox-edit.js [hunk 2]
```diff
@@ -550,6 +545,39 @@
...
+							<x title='remove tag' class='tagify__tag__removeBtn'></x>
+							<div>
+								<span class='tagify__tag-text'>${tagLabel}</span>
+								${tagData['data-actions'] ?
+									`<a href="javascript:;" class="tagify__filter-icon" ${this.getAttributes(tagData)}></a>` : ''
+								}
+							</div>
+						</tag>`
+					}
+					catch(err){}
```
Comment (preda-bogdan) [inline includes/views/js/import-metabox-edit.js]: Maybe add a log here instead for the error.

**Review by preda-bogdan — APPROVED** [review]
I will approve this even though I left a few small changes that maybe you can consider before proceeding.

**Review by GrigoreMihai — APPROVED** [review]
I do not see other issues apart from what Bogdan mentioned.

**Review by GrigoreMihai — APPROVED** [review]

**Pull request merged**